CMakeLists.txt

```diff
@@ -420,13 +431,16 @@ SET (SYSTEM_LIBRARIES "-Wl,--as-needed" ${SYSTEM_LIBRARIES})
 ENDIF (HAVE_LINKER_AS_NEEDED)
 
 ADD_SUBDIRECTORY (src)
-ADD_SUBDIRECTORY (ext)
-ADD_SUBDIRECTORY (tools)
-IF (ENABLE_PERL OR ENABLE_PYTHON OR ENABLE_RUBY OR ENABLE_TCL)
-    ADD_SUBDIRECTORY (bindings)
-ENDIF (ENABLE_PERL OR ENABLE_PYTHON OR ENABLE_RUBY OR ENABLE_TCL)
-ADD_SUBDIRECTORY (examples)
-ADD_SUBDIRECTORY (doc)
+
+IF (NOT MSVC)
+  ADD_SUBDIRECTORY (ext)
+  ADD_SUBDIRECTORY (tools)
+  IF (ENABLE_PERL OR ENABLE_PYTHON OR ENABLE_RUBY OR ENABLE_TCL)
+      ADD_SUBDIRECTORY (bindings)
+  ENDIF (ENABLE_PERL OR ENABLE_PYTHON OR ENABLE_RUBY OR ENABLE_TCL)
+  ADD_SUBDIRECTORY (examples)
+  ADD_SUBDIRECTORY (doc)
+ENDIF ()
```

Comment:
Why are we saying none of these can be built with MSVC? Is that actually true?

Reply:
Building the `ext` seemed to be quite hard due to fncntl ... I don't have the time + knowledge to look into it right now, unfortunately. I think doing a first, small PR to get something to build successfully on Windows is a better approach.

Reply:
Hmm, I am taking this back. I'll try to get ext to build on Windows as well. seems like just a replacement for `sys/stat.h` is needed, and that seems to be `direct.h` on Windows.

Reply:
ext is building on windows now, tests are running and the default tools are available, too.